pkg/process/subscribers/gpu_subscriber.go

```diff
@@ -0,0 +1,138 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package subscribers implements subscribers to workloadmeta.
+package subscribers
+
+import (
+	"context"
+	"sync/atomic"
+
+	tagger "github.com/DataDog/datadog-agent/comp/core/tagger/def"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/util/common"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// GPUSubscriber is a subscriber that listens for GPU events from workloadmeta
+type GPUSubscriber struct {
+	gpuDetected atomic.Bool
+	gpuEventsCh chan workloadmeta.EventBundle
+	stopCh      chan struct{}
+	wmeta       workloadmeta.Component
+	tagger      tagger.Component
+}
+
+// NewGPUSubscriber creates a new GPUDetector instance
+func NewGPUSubscriber(wmeta workloadmeta.Component, tagger tagger.Component) *GPUSubscriber {
+	return &GPUSubscriber{
+		stopCh: make(chan struct{}),
+		wmeta:  wmeta,
+		tagger: tagger,
+	}
+}
+
+// Run starts the GPU detector, which listens for workloadmeta events to detect GPUs on the host
+func (g *GPUSubscriber) Run(_ context.Context) error {
+	log.Info("Starting GPU subscriber")
+	filter := workloadmeta.NewFilterBuilder().
+		SetSource(workloadmeta.SourceRuntime).
+		SetEventType(workloadmeta.EventTypeSet).
+		AddKind(workloadmeta.KindGPU).
+		Build()
+
+	g.gpuEventsCh = g.wmeta.Subscribe(
+		"gpu-subscriber-set-gpu",
+		workloadmeta.NormalPriority,
+		filter,
+	)
+	go func() {
+		for {
+			select {
+			case eventBundle, ok := <-g.gpuEventsCh:
+				if !ok {
+					return
+				}
+				g.processEvents(eventBundle)
+				eventBundle.Acknowledge()
+			case <-g.stopCh:
+				g.wmeta.Unsubscribe(g.gpuEventsCh)
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+// IsGPUDetected checks if a GPU has been detected
+func (g *GPUSubscriber) IsGPUDetected() bool {
+	return g.gpuDetected.Load()
+}
+
+// SetGPUDetected sets the GPU detected status
+func (g *GPUSubscriber) SetGPUDetected(value bool) {
+	g.gpuDetected.Store(value)
+}
+
+// processEvents processes the events received from workloadmeta
+func (g *GPUSubscriber) processEvents(eventBundle workloadmeta.EventBundle) {
+	for _, event := range eventBundle.Events {
+		_, ok := event.Entity.(*workloadmeta.GPU)
+		if !ok {
+			log.Debugf("Expected workloadmeta.GPU got %T, skipping", event.Entity)
+			continue
+		}
+
+		g.SetGPUDetected(true)
+		break
```

Comment:
maybe you can just simplify this for the happy path
```suggestion
		if _, ok := event.Entity.(*workloadmeta.GPU); ok {
			g.SetGPUDetected(true)
			break
		}
```